bindings/python/cconfigspace/configuration.py

```diff
@@ -44,5 +57,28 @@ def configuration_space(self):
     return self._configuration_space
 
   def check(self):
+    """Check the validity of the current configuration based on the generative configuration space.
+    """
     res = ccs_configuration_check(self.handle)
     Error.check(res)
+
+  @property
+  def hyperparameters(self):
+    """Returns the list of hyperparameters of the current configuration."""
+    return self.configuration_space.hyperparameters
+
+  def __getitem__(self, idx):
```

Comment:
Why not alias `__getitem__` and `value` in binding.py?
If you want to favor the index code path we can think about switching the isinstance conditions around.